src/ncp/spinel.c

```diff
@@ -332,7 +331,14 @@ spinel_datatype_vunpack_(const uint8_t *data_ptr, spinel_size_t data_len, const
 
             if (arg_ptr)
             {
-                *arg_ptr = (spinel_ipv6addr_t *)data_ptr;
+                if (*arg_ptr)
+                {
+                    memcpy(*arg_ptr, data_ptr, sizeof(spinel_ipv6addr_t));
+                }
+                else
+                {
+                    *arg_ptr = (spinel_ipv6addr_t *)data_ptr;
+                }
```

Comment:
Thanks for sharing your concerns. However, I think it's non-initialized pointers that is error-prone and dangerous. We should keep the practice to initialize a pointer to null if there's not a exact address for it, not only for the spinel API, but also other places.

This PR did have impact on other projects, that is, requiring pointers passing to spinel parser to be initialized to null. I don't think this impact is bad, indeed. And it doesn't take very much efforts to add initialization.

I did consider providing a wrapper before submitting this PR, however, I didn't figure out how to deal with the variable arguments. Is there a clean way to build a `va_list` from another? And if not using variable arguments, it will result in a lot of code. I like the simplicity of the spinel API and I prefer to keep it simple.

Thoughts? @abtink